core/shared/src/main/scala/scalaz/zio/internal/SingleThreadedRingBuffer.scala

```diff
@@ -0,0 +1,53 @@
+package scalaz.zio.internal
+
+private[zio] final class SingleThreadedRingBuffer[A <: AnyRef](capacity: Int) {
```

Comment:
A couple questions regarding semantics of this class:
1. Is it intended that elements get overwritten on buffer wrap around?
2. SingleThreaded = NotThreadSafe?
3. `pop` returning Unit is unusual. Could you elaborate?

All in all, it would be nice to add a scaladoc to this class documenting its semantics. Could you?

Reply:
1. Yes, it's a ring buffer
2. Yes and it's fine for the purpose. Other fiber structures are the same.
3. It's not 'pop', it's 'remove head'.

Reply:
2. We have doubts about thread safety here, however, both `Stack` & `StackBool` are currently implemented the same way, so I guess we'll find out sooner or later...